src/imitation/scripts/compare_to_baseline.py

```diff
@@ -0,0 +1,100 @@
+"""Compare experiment results to baseline results.
+
+This script compares experiment results to the results reported in the
+[paper](https://arxiv.org/pdf/2211.11972.pdf). It takes as input a CSV file
+containing experiment results, and outputs a table of p-values comparing the experiment
+results to the baseline results.
+
+Usage:
+    $ python compare_to_baseline.py <path_to_results_file>
+
+The results file should be a CSV file containing the following columns:
+    * algo: The name of the imitation algorithm.
+    * env_name: The name of the environment.
+    * imit_return_summary: A string containing the mean and standard deviation of the
+      experiment returns, as reported by `imitation.scripts.analyze`.
+"""
+
+import numpy as np
+import pandas as pd
+import scipy
+
+from imitation.data import types
+
+
+def compare_results_to_baseline(results_file: types.AnyPath) -> str:
+    """Compare benchmark results to baseline results.
+
+    Args:
+        results_file: Path to a CSV file containing experiment results.
+
+    Returns:
+        A string containing a table of p-values comparing the experiment results to
+        the baseline results.
+    """
+    data = pd.read_csv(results_file)
+    data["imit_return"] = data["imit_return_summary"].apply(
+        lambda x: float(x.split(" ")[0]),
+    )
+    summary = (
+        data[["algo", "env_name", "imit_return"]]
+        .groupby(["algo", "env_name"])
+        .describe()
+    )
+    summary.columns = summary.columns.get_level_values(1)
+    summary = summary.reset_index()
+
+    # Table 2 (https://arxiv.org/pdf/2211.11972.pdf)
+    # todo: store results in this repo outside this file
+    baseline = pd.DataFrame.from_records(
+        [
+            {
+                "algo": "??exp_command=bc",
```

Comment:
For whatever reason this is how the analyze script stores the algo field

Reply:
Hmm, we should probably try to clean that up! @taufeeque9 any idea why it's doing this?

Reply:
That is the default behavior (else statement below) when the Sacred command is not recognized. The commands to train `bc` and `dagger` was changed earlier without updating the `_get_algo_name` function in `analyze.py`. A check for `preference_comparisons` is also missing from the function.

Function reproduced from `analyze.py`:
```python
def _get_algo_name(sd: sacred_util.SacredDicts) -> str:
    exp_command = _get_exp_command(sd)

    if exp_command == "gail":
        return "GAIL"
    elif exp_command == "airl":
        return "AIRL"
    elif exp_command == "train_bc":
        return "BC"
    elif exp_command == "train_dagger":
        return "DAgger"
    else:
        return f"??exp_command={exp_command}"
```